### PR TITLE
Support for dotnet

### PR DESCRIPTION
**Pull request by victorhurdugaci**
- Support for dotnet in KoreBuild
- New package name: KoreBuild-dotnet
- Tests don't work yet
- Always builds with the latest dotnet
- No branch support yet

please review @pranavkm @anurse

### CODE REVIEW

**Comment by victorhurdugaci** [comment]
Part of https://github.com/aspnet/Universe/issues/324

File: KoreBuild-dotnet/build/install.ps1 [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#
```
Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/install.ps1]: This file is copied from CLI so ignore it

File: KoreBuild-dotnet/build/install.sh [hunk 2]
```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env sh
```
Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/install.sh]: This file is copied from CLI so ignore it

**Comment by victorhurdugaci** [comment]
@BrennanConroy

File: KoreBuild-dotnet/build/_dotnet-restore.shade [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    Restores nuget packages required for dotnet projects. Downloads and executes dotnet sdk tools.
+
+restoreDir=''
+    Optional. The directory in which to execute the dnu restore command.
+*/}
+
+default currentDir = '${ Directory.GetCurrentDirectory() }'
+default restoreDir = '${ currentDir }'
+
+default restore_options=' ${ E("KOREBUILD_DNU_RESTORE_OPTIONS") } ${ IsLinux ? string.Empty : "--parallel" }'
```
Comment (BrennanConroy) [inline KoreBuild-dotnet/build/_dotnet-restore.shade]: DNU?

File: KoreBuild-dotnet/build/install.cmd [hunk 4]
```diff
@@ -1,10 +1,10 @@
 @Echo off
 
-for /f "delims=" %%i in ('PowerShell -NoProfile -NoLogo -ExecutionPolicy unrestricted -Command "[System.IO.Path]::GetTempFileName()"') do set DNVM_CMD_PATH_FILE=%%i.cmd
+for /f "delims=" %%i in ('PowerShell -NoProfile -NoLogo -ExecutionPolicy unrestricted -Command "[System.IO.Path]::GetTempFileName()"') do set INSTALL_CMD_PATH_FILE=%%i.cmd
```
Comment (BrennanConroy) [inline KoreBuild-dotnet/build/install.cmd]: Put quotes around %%i.cmd

File: build-template-dotnet/build.cmd [hunk 5]
```diff
@@ -13,28 +16,27 @@ IF NOT EXIST %LocalAppData%\NuGet md %LocalAppData%\NuGet
...
-:restore
-IF EXIST packages\Sake goto getdnx
-IF "%BUILDCMD_KOREBUILD_VERSION%"=="" (
-    .nuget\nuget.exe install KoreBuild -ExcludeVersion -o packages -nocache -pre
-) ELSE (
-    .nuget\nuget.exe install KoreBuild -version %BUILDCMD_KOREBUILD_VERSION% -ExcludeVersion -o packages -nocache -pre
-)
+:getsake
+IF EXIST packages\Sake goto skipgetsake
 .nuget\NuGet.exe install Sake -ExcludeVersion -Source https://www.nuget.org/api/v2/ -Out packages
```
Comment (BrennanConroy) [inline build-template-dotnet/build.cmd]: nuget v3?

File: KoreBuild-dotnet/build/install.cmd [hunk 6]
```diff
@@ -1,10 +1,10 @@
 @Echo off
```
Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/install.cmd]: This file is from the CLI. Don't comment on it

File: KoreBuild-dotnet/build/install.cmd (hunk 4 above)
Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/install.cmd]: File is from CLI

Comment (BrennanConroy) [inline KoreBuild-dotnet/build/install.cmd]: Where? I need to tell them its wrong

Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/install.cmd]: I lied. This is our old `dnvm.cmd` file: https://github.com/aspnet/Universe/blob/dev/build/dnvm.cmd it's been like this for ages

Comment (BrennanConroy) [inline KoreBuild-dotnet/build/install.cmd]: Change it

File: KoreBuild-dotnet/build/_dotnet-restore.shade (hunk 3 above)
Comment (muratg) [inline KoreBuild-dotnet/build/_dotnet-restore.shade]: Yeah, aren't we going to change the env var names?

Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/_dotnet-restore.shade]: Forgot that one

**Comment by victorhurdugaci** [comment]
Updated.

@cesarbs fyi

**Comment by anurse** [comment]
Psh, `KoreBuild-dotnet`, I prefer `DoreBuild`, or if we want to keep the `k` around: `DorkBuild` ;)

**Comment by muratg** [comment]
@anurse you :trollface:

**Comment by victorhurdugaci** [comment]
So, should I renamed it to `DorkBuild` or merge it?

**Comment by BrennanConroy** [comment]
Definitely __*DorkBuild*__

File: KoreBuild-dotnet/build/_dotnet-build.shade [hunk 7]
```diff
@@ -36,8 +36,8 @@ default build_options=' ${E("KOREBUILD_DNU_BUILD_OPTIONS")}'
         }
 
         var projectsArg=projectFile.Replace(";", " ");
-        var dnuArgs=string.Format("build{0} {1} --configuration {2}", build_options, projectsArg, configuration);
-        Dnu(dnuArgs);
+        var dotnetArgs=string.Format("build{0} {1} --configuration {2}", build_options, projectsArg, configuration);
```
Comment (anurse) [inline KoreBuild-dotnet/build/_dotnet-build.shade]: `dotnet-compile`?

File: KoreBuild-dotnet/build/_dotnet.shade [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+command=''
+    The `dotnet` subcommand to execute.
+dotnetDir=''
+    Optional. The directory in which to execute the `dotnet` command.
+*/}
+
+default currentDir = '${Directory.GetCurrentDirectory()}'
+default dotnetDir = '${ currentDir }'
+
+exec program='cmd' commandline='/C dotnet ${command}' workingdir='${dotnetDir}' if='!IsLinux'
```
Comment (anurse) [inline KoreBuild-dotnet/build/_dotnet.shade]: Shouldn't be necessary. `dotnet` is a real exe on Windows.

File: KoreBuild-dotnet/build/install.sh (hunk 2 above)
Comment (anurse) [inline KoreBuild-dotnet/build/install.sh]: image

Comment (anurse) [inline KoreBuild-dotnet/build/install.sh]: I mean: OK :trollface:

**Comment by anurse** [comment]
Just leave it as `KoreBuild-dotnet`... but I still refuse to call it that and will refer to it solely as `DorkBuild` from now on.

:shipit:

File: KoreBuild-dotnet/build/_dotnet-build.shade (hunk 7 above)
Comment (victorhurdugaci) [inline KoreBuild-dotnet/build/_dotnet-build.shade]: Yup. Fixing it now

**Pull request merged**